### PR TITLE
Implementation of max0() using IntrinsicFunction

### LINKED ISSUE #1739: [SciPy] [Fitpack]: AssertFailed `args.size() == (func->n_args - offset)`
```fortran
      subroutine fppasu()
      integer max0
      mm = max0(1,2)

      nplu = max0(1,2/2,1)

      end
```

> LFortran

```console
% lfortran --fixed-form --implicit-typing --implicit-interface --show-asr /Users/pgoswami/repos/scipy/scipy/interpolate/fitpack/fppasu.f
LCOMPILERS_ASSERT failed: /Users/pgoswami/Desktop/lfortran/src/libasr/asr_utils.h
function set_absent_optional_arguments_to_null(), line number 1986 at 
args.size() == (func->n_args - offset)
zsh: abort      lfortran --fixed-form --implicit-typing --implicit-interface --show-asr 
```

### PR DESCRIPTION
**Pull request by gptsarthak**
Will fix #1739
Work in Progress

The ```max0``` and ```min0``` Fortran functions can accept any arbitrary number of arguments.
@czgdp1807 Can you please review this and guide me with the implementation of the  ```instantiate_Max()``` function? How would I write that?

### CODE REVIEW

**Review by czgdp1807 — COMMENTED** [review]
Please add a test. Regarding instantiate @certik, see how `instantiate_Any`, `instantiate_Sum` is implemented, from there you can learn to write `instantiate_Max`.

**Comment by Thirumalai-Shaktivel** [comment]
Yup, `instantiate_*` is used to create a new function in the global scope (in ASR) and return functioncall (to the newly created function) expression. So, in the function as @certik said do `create a loop and go over all arguments and find the maximum, using ASR`. The `instantiate_functions` and `instantiate_Abs` seems to be the simplest one, check it out.

**Comment by Pranavchiku** [comment]
@gptsarthak are you working on it?

**Comment by certik** [comment]
@Pranavchiku go ahead and finish it. @gptsarthak you can work on the format, that way others do not depend on your work immediately, so everybody can work as time allows.

**Comment by gptsarthak** [comment]
Really sorry if this was a blocker. I have made a new PR #1799  with a temporary fix. With its help 3 and 6 argument max0 will work
I will push the test and matching function signature fix here. If given a chance, it would be helpful if @Pranavchiku can help me with ```instantiate_max0```, and I'll try my best to complete the rest of the PR asap.
In the meantime, I'll start figuring out string formatting.

**Comment by certik** [comment]
No worries, thanks for your work, the plan you outlined looks good.

**Comment by Pranavchiku** [comment]
Very much thanks to @Thirumalai-Shaktivel for helping me here. I now have clear idea and will complete this PR soon.

**Comment by Pranavchiku** [comment]
@certik this PR can have a final review.

File: integration_tests/intrinsics_52.f90 [hunk 1]
```diff
@@ -13,10 +13,4 @@ program intrinsics_52
 if (max0(a,b,c) /= c) error stop
 if (max0(d,e,f,a,b,c) /= e) error stop
 
-if (min0(10,20) /= 10) error stop
-if (min0(3,5,4) /= 3) error stop
-if (min0(a,b) /= b) error stop
-if (min0(c,b,a) /= b) error stop
-if (min0(d,e,f,a,b,c) /= f) error stop
```
Comment (certik) [inline integration_tests/intrinsics_52.f90:20]: Why were these removed?

**Comment by certik** [comment]
Otherwise I think this looks good.

File: integration_tests/intrinsics_52.f90 (hunk 1 above)
Comment (Pranavchiku) [reply integration_tests/intrinsics_52.f90:20]: I am not sure, I will separate tests for min0 and max0

File: integration_tests/CMakeLists.txt [hunk 2]
```diff
@@ -439,8 +439,9 @@ RUN(NAME intrinsics_48 LABELS gfortran llvm NOFAST)
 RUN(NAME intrinsics_49 LABELS gfortran llvm)
 RUN(NAME intrinsics_50 LABELS gfortran llvm NOFAST)
 RUN(NAME intrinsics_51 LABELS gfortran llvm NOFAST)
-RUN(NAME intrinsics_52 LABELS gfortran llvm) # min0, max0
+RUN(NAME intrinsics_52 LABELS gfortran)
```
Comment (certik) [inline integration_tests/CMakeLists.txt:442]: Why was `llvm` removed here? I think that is a regression?

**Comment by certik** [comment]
Otherwise I think it looks good.

File: integration_tests/CMakeLists.txt (hunk 2 above)
Comment (Pranavchiku) [reply integration_tests/CMakeLists.txt:442]: I was actually wondering that, but then thought it will be added once `min0` gets implemented.

**Comment by Pranavchiku** [comment]
@gptsarthak @HarshitaKalani any of you can go ahead and implement `min0`. You can follow the files changed in this PR and make the necessary changes. This will be easy to do and will help to get an idea about intrinsic architecture which will further help in implementation of `maxloc`, `minloc`

**Comment by HarshitaKalani** [comment]
Done at https://github.com/lfortran/lfortran/pull/1898
